### ISSUE TITLE
Other indexing error: decoding to str: need a bytes-like object, float found

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#2570]
```
Dec  3 01:31:38 intlab-010 journal[3444137]: 2021-12-03T01:31:38.506260 ERROR 3444137 140538020952960 pbench-index-re.pbench-index main 351 -- Other indexing error: decoding to str: need a bytes-like object, float found
Traceback (most recent call last):
  File "/opt/pbench-server/bin/pbench-index.py", line 326, in main
    idxctx.es, actions, fp, idxctx.logger, idxctx._dbg
  File "/opt/pbench-server/lib/pbench/indexer.py", line 762, in es_index
    for ok, resp_payload in streaming_bulk_generator:
  File "/opt/pbench-server/lib/python3.8/site-packages/elasticsearch1/helpers/__init__.py", line 159, in streaming_bulk
    for bulk_actions in _chunk_actions(actions, chunk_size, max_chunk_bytes, client.transport.serializer):
  File "/opt/pbench-server/lib/python3.8/site-packages/elasticsearch1/helpers/__init__.py", line 53, in _chunk_actions
    for action, data in actions:
  File "/opt/pbench-server/lib/pbench/indexer.py", line 714, in actions_tracking_closure
    for cl_action in cl_actions:
  File "/opt/pbench-server/lib/pbench/indexer.py", line 4259, in make_all_actions
    for action in self.mk_result_data_actions():
  File "/opt/pbench-server/lib/pbench/indexer.py", line 4649, in mk_result_data_actions
    for source, source_id, parent_id, doc_type in sources:
  File "/opt/pbench-server/lib/pbench/indexer.py", line 1691, in _make_source_json
    iter_data, iter_name, iter_number, result_json
  File "/opt/pbench-server/lib/pbench/indexer.py", line 1835, in _handle_iteration
    bm_md["trafficgen_uid_tmpl"], bm_md, run=self.run_metadata
  File "/opt/pbench-server/lib/pbench/indexer.py", line 1899, in expand_template
    s = re.sub(m, val, s)
  File "/usr/lib64/python3.6/re.py", line 191, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/usr/lib64/python3.6/re.py", line 326, in _subx
    template = _compile_repl(template, pattern)
  File "/usr/lib64/python3.6/re.py", line 317, in _compile_repl
    return sre_parse.parse_template(repl, pattern)
  File "/usr/lib64/python3.6/sre_parse.py", line 879, in parse_template
    s = Tokenizer(source)
  File "/usr/lib64/python3.6/sre_parse.py", line 227, in __init__
    string = str(string, 'latin1')
TypeError: decoding to str: need a bytes-like object, float found
```

**Comment by portante** [comment]
Fixed on `main` proposed with #3029.

**Comment by portante** [comment]
Fixed on `b0.69` via #3032.

Note that this server fix will not be back-ported to `v0.71` since that branch does not contain a deployable server.
